src/bootstrap/tool.rs

```diff
@@ -562,10 +604,22 @@ impl Step for Rustdoc {
             features.as_slice(),
         );
 
-        builder.info(&format!(
-            "Building rustdoc for stage{} ({})",
-            target_compiler.stage, target_compiler.host
-        ));
+        let msg = if self.compiler.host == target {
+            format!(
+                "Building rustdoc (stage{} -> stage{})",
+                target_compiler.stage - 1,
+                target_compiler.stage
+            )
+        } else {
+            format!(
+                "Building rustdoc (stage{}:{} -> stage{}:{})",
+                target_compiler.stage - 1,
+                self.compiler.host,
+                target_compiler.stage,
+                target
+            )
+        };
+        builder.info(&msg);
```

Comment:
could this also use `tooling_output`? that would also avoid the inconsistency between rustdoc and clippy ("tool clippy" vs "rustdoc"):
```
; x build --dry-run rustdoc clippy
Building bootstrap
    Finished dev [unoptimized] target(s) in 0.07s
Building stage0 library artifacts (aarch64-apple-darwin) 
Building compiler artifacts (stage0 -> stage1)
Assembling stage1 compiler
Building rustdoc (stage0 -> stage1)
Building stage1 library artifacts (aarch64-apple-darwin) 
Building compiler artifacts (stage1 -> stage2)
Building tool clippy-driver (stage1 -> stage2)
```

(also, *fuck yeah*, this shows that rustdoc and clippy are behaving differently in the output!!!!)